LiveSched/src/main/java/dev/coms4156/project/livesched/Task.java

```diff
@@ -0,0 +1,117 @@
+package dev.coms4156.project.livesched;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+/**
+ * Represents a task that has to be done. 
+ * This class stores the id, resources needed, and the location of the task.
+ */
+public class Task {
+  private final String taskId;
+  private Map<ResourceType, Integer> resourceList; // Key = ResoureceType, Value = Units needed
+  private int priority; 
+  private LocalDateTime startTime;
+  private LocalDateTime endTime;
+  private double latitude;
+  private double longitude;
+
+  /**
+   * Constructs a new Task object with the given parameters.
+   *
+   * @param taskId        the unique ID of the resource
+   * @param resourceList  the map of resources and their quantity needed
+   * @param priority      the priority of the task
+   * @param endTime       the time that task ends
+   * @param latitude      the latitude of the resource's location
+   * @param longitude     the longitude of the task's location
+   */
+  public Task(String taskId, Map<ResourceType, Integer> resourceList, int priority, 
+        LocalDateTime endTime, double latitude, double longitude) {
+    this.taskId = taskId;
+    this.resourceList = resourceList;
+    this.priority = priority;
+    this.startTime = LocalDateTime.now(); // Initially starting now
+    this.endTime = endTime;
+    this.latitude = latitude;
+    this.longitude = longitude;
+  }
+
+  /**
+   * Set the start time of the task.
+   *
+   * @param startTime the time that task starts
+   */
+  public void setStartTime(LocalDateTime startTime) {
+    this.startTime = startTime;
+  }
+
+  /**
+   * Updates the end time of the task.
+   *
+   * @param endTime the time that task ends
+   */
+  public void updateEndTime(LocalDateTime endTime) {
+    this.endTime = endTime;
+  }
+
+  /**
+   * Adds a resource needed for the task.
+   *
+   * @param resourceType  the type of resource needed
+   * @param quantity      the quantity of the resource needed
+   */
+  public void addResource(ResourceType resourceType, int quantity) {
+    resourceList.put(resourceType, quantity);
+  }
+
+  /**
+   * Chekcs if all resources needed for the task is available.
+   *
+   * @return true if all resources are available, false otherwise
+   */
+  public boolean checkResourceAvailability() {
```

Comment:
getAvailableUnits() counts the number of currently available units. (I should go back and update the name actually.) So, if we are doing immediate scheduling, I think it makes sense to check the availability this way. But, if we are going to allow future scheduling, this method would unfortunately not work. Also, I think this could be handled by the Schedule class, like checking and assigning resources to the task at the same time. What do you guys think?

Reply:
On second thought, it might actually be helpful to check if all resources are available before actually assigning them to avoid partial assignment of resources (depending on whether we want to allow this or not).

Reply:
Yes, I think it does make sense for this to be. handled by the Schedule class! I've removed this function from this class in my recent commit.